include/exadg/incompressible_navier_stokes/user_interface/parameters.cpp

```diff
@@ -500,6 +541,10 @@ Parameters::check(dealii::ConditionalOStream const & pcout) const
       use_cell_based_face_loops == true,
       dealii::ExcMessage(
         "Cell based face loops have to be used for matrix-free implementation of block diagonal preconditioner."));
+
+    AssertThrow(spatial_discretization == SpatialDiscretization::L2,
```

Comment:
Here, I would also assert that the type of discretization is not Hdiv, with the exception message "not implemented.".